cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/FieldLineageAdmin.java

```diff
@@ -389,10 +391,11 @@ private Set<Field> filter(String prefix, Set<Field> fields) {
    * Retrieves a summary of all endpoints in a given program run.
    *
    * @param namespaceId program namespace
-   * @param programRunId start time of the run in millis corresponding to the id of a program run
+   * @param programReference start time of the run in millis corresponding to the id of a program run
+   * @param runId run ID which is a UUID
    * @return Summary of all Endpoints in the particular run of the pipeline.
    */
-  public EndpointsSummary getEndpoints(String namespaceId, ProgramRunId programRunId) {
-    return new EndpointsSummary(fieldLineageReader.getEndpoints(namespaceId, programRunId));
+  public EndpointsSummary getEndpoints(String namespaceId, ProgramReference programReference, String runId) {
```

Comment:
it's better to use the RunId class instead of a string, it will ensure that the runid is well-formed. This is especially important since the runid is user provided.